lib/GADS.pm

```diff
@@ -2753,6 +2755,290 @@ prefix '/:layout_name' => sub {
         template 'data' => $params;
     };
 
+    prefix '/report' => sub {
+
+        #view all reports for this instance
+        get '' => require_login sub {
+            my $user   = logged_in_user;
+            my $layout = var('layout') or pass;
+
+            if ( app->has_hook('plugin.linkspace.data_before_request') ) {
+                app->execute_hook( 'plugin.linkspace.data_before_request',
+                    user => $user );
+            }
+
+            my %params = (
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $records = GADS::Records->new(%params);
+
+            my $alert = GADS::Alert->new(
+                user   => $user,
+                layout => $layout,
+                schema => schema,
+            );
+
+            my $base_url = request->base;
+
+            my $params;
+
+            $params->{alerts}      = $alert->all;
+            $params->{header_type} = 'table_tabs';
+
+            $params->{layout_obj} = $layout;
+            $params->{layout}     = $layout;
+
+            $params->{header_back_url} = "${base_url}table";
+            $params->{breadcrumbs}     = [
+                Crumb( $base_url . "table/", "Tables" ),
+                Crumb( "",                   "Table: " . $layout->name )
+            ];
+
+            my $layout_id = $layout->{instance_id}; 
+
+            my $reports = schema->resultset('Report')->load_all_reports($layout_id);
+
+            $params->{viewtype} = 'table';
+            $params->{reports}  = $reports;
+
+
+            template 'report' => $params;
+        };
+
+        #add a report
+        any [ 'get', 'post' ] => '/add' => require_login sub {
+            if ( body_parameters && body_parameters->{submit} ) {
+                my $layout             = var('layout') or pass;
+                my $report_description = body_parameters->{report_description};
+                my $report_name        = body_parameters->{report_name};
+                my $checkbox_fields_full = body_parameters->{checkbox_fields};
```

Comment:
Should be able to use: `my @checkbox_fields = body_parameters->get_all('checkbox_fields')` instead to get all values.

In general, it's better to use `body_parameters->get('foo')` instead of `body_parameters->{foo}`